app/services/gift_stickers_service.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module GiftStickersService
+  module_function
+
+  def call
+    user_dates = []
+
+    User.where(state: 'incompleted').find_in_batches do |group|
+      group.each do |u|
+        date = u.receipt.min_by do |receipt|
```

Comment:
`u.receipt.min_by` will return the first pull request submitted by the user, we should be looking at the date when the last pull request was submitted.

In the case of a user with three pull requests, the date on the latest pull request is the date that they achieved the score of 3. Users who scored three earlier should be awarded gifts first.